### PR TITLE
Update workflow to completed if PC.status is OK

### LINKED ISSUE #827: (PC) When status is changed to OK_STATUS set preservationAuditWF to "completed"
- Once we change a preserved_copy status to `OK_STATUS`, we need to remove the preservationAuditWF error. 
- So, clear the workflows error so it no longer appears in argo

### PR DESCRIPTION
**Pull request by SaravShah**
- This is a pretty long PR, so if you would like me to break it up please let me know
- Best to review this PR commit by commit
- last 2 commits are refactor commits, changing name of class and class method names in `WorkflowReporter` and switch the context block to describe block, for our RSpec tests. 

needs to deal with M2C and C2M still but created a story time question for that here -> (https://docs.google.com/document/u/1/d/10HMhTMWlZaM1DJChFr2r3coOVDZKe0EDAH8PPumb-2E/edit#) 

fixes #827

### CODE REVIEW

File: app/services/workflow_reporter.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+# send errors to preservationAuditWF workflow for an object via ReST calls.
+class WorkflowReporter
+
+  DOR = 'dor'.freeze
+  PRESERVATIONAUDITWF = 'preservationAuditWF'.freeze
+  NO_WORKFLOW_HOOKUP = 'no workflow hookup - assume you are in test or dev environment'.freeze
+  # this method will always return true because of the dor-workflow-service gem
```
Comment (jmartin-sul) [inline app/services/workflow_reporter.rb]: very nitpicky, but i'd sorta like a blank line between the last constant and the method comment. mildly surprised rubocop didn't complain.

File: app/services/workflow_reporter.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if Settings.workflow_services_url.present?
+      Dor::WorkflowService.update_workflow_error_status(DOR, "druid:#{druid}", PRESERVATIONAUDITWF, process_name, error_message)
+    else
+      Rails.logger.warn(NO_WORKFLOW_HOOKUP)
+    end
+  end
+
+  def self.report_completed(druid, process_name)
+    if Settings.workflow_services_url.present?
+      Dor::WorkflowService.update_workflow_status(DOR, "druid:#{druid}", PRESERVATIONAUDITWF, process_name, 'completed')
```
Comment (jmartin-sul) [inline app/services/workflow_reporter.rb]: if the other strings that don't change are becoming constants, i'd somewhat prefer to do `'completed'` also.

File: app/services/workflow_errors_reporter.rb [hunk 3]
```diff
@@ -1,5 +1,4 @@
-require 'faraday'
-require 'retries'
```
Comment (ndushay) [inline app/services/workflow_errors_reporter.rb]: wow - we missed these defunct requires before?  oops.

**Comment by SaravShah** [comment]
appreciate it @jmartin-sul

**Pull request merged**